pkg/kv/kvserver/store_create_replica.go

```diff
@@ -249,28 +249,25 @@ func fromReplicaIsTooOldRLocked(toReplica *Replica, fromReplica *roachpb.Replica
 	return !found && fromReplica.ReplicaID < desc.NextReplicaID
 }
 
-// addToReplicasByKeyLocked adds the replica to the replicasByKey btree. The
-// replica must already be in replicasByRangeID. Requires that Store.mu is held.
-//
-// Returns an error if a different replica with the same range ID, or an
-// overlapping replica or placeholder exists in this Store.
-func (s *Store) addToReplicasByKeyLocked(repl *Replica) error {
+// addToReplicasByKeyLockedReplicaRLocked adds the replica to the replicasByKey
+// btree. The replica must already be in replicasByRangeID. Returns an error if
+// a different replica with the same range ID, or an overlapping replica or
+// placeholder exists in this Store. Replica.mu must be at least read-locked.
+func (s *Store) addToReplicasByKeyLockedReplicaRLocked(repl *Replica) error {
 	if !repl.IsInitialized() {
```

Comment:
Now that we're holding `repl.mu` here, it tingles my spidey senses to call `IsInitialized` here. I checked and it doesn't acquire the lock - it loads an atomic - but I bet it does this because we had deadlocks in the past. Maybe here we can entertain the more obviously correct `repl.descRLocked().IsInitialized()`? You call `descRLocked` below anyway.